### ISSUE TITLE
repeated TAIT doesn't detect lifetime conflict

### ISSUE THREAD

**Issue by nikomatsakis** [rust-lang/rust#86465]
This code compiles, but it should not:

```rust
#![feature(min_type_alias_impl_trait)]

type X<'a, 'b> = impl std::fmt::Debug;

fn f<'t, 'u>(a: &'t u32, b: &'u u32) -> (X<'t, 'u>, X<'u, 't>) {
    (a, a)
}
```

The function creates two constraints:

* `X<'t, 'u> = &'t u32`, which means `type X<'a, 'b> = &'a u32`
* `X<'u, 't> = &'t u32`, which means `type X<'a, 'b> = &'b u32`

These should not both be satisfiable!

**Comment by nikomatsakis** [comment]
@rustbot assign @nikomatsakis 

I have been working on this. It might also make sense to assign to @oli-obk instead though.

**Comment by nikomatsakis** [comment]
In progress PR: https://github.com/rust-lang/rust/pull/86410
